packages/nextjs-kit/src/components/paginator.tsx

```diff
@@ -0,0 +1,257 @@
+          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 ${
+            currentPageQuery === pageNumber ? 'border-blue-700 border-2' : ''
+          }
+          `}
+          onClick={handlePageClick}
+          key={pageNumber}
+        >
+          {pageNumber}
+        </button>
+      );
+
+      // separator button
+      if (i === breakStart) {
+        if (isNumber(breakAdd)) {
+          if (breakStart + breakAdd >= totalPages) {
+            buttons.push(defaultButton);
+            continue;
+          }
+        }
+        buttons.push(
+          <button
+            className={`hidden md:block h-16 w-12 border-2 border-black bg-slate-200 hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300"
+            }`}
+            onClick={() => {
+              if (isNumber(breakAdd)) {
+                setBreakStart(breakStart + breakAdd);
+              }
+            }}
+            key={'...'}
+          >
+            ...
+          </button>
+        );
+      }
+      // if we have a breakStart, don't render the middle buttons
+      if (isNumber(breakStart) && isNumber(breakEnd)) {
+        if (pageNumber >= breakStart && pageNumber < breakEnd) {
+          if (isNumber(windowWidth)) {
+            if (windowWidth < 768) {
+              buttons.push(defaultButton);
+            }
+            continue;
+          }
+        }
+      }
+      buttons.push(defaultButton);
+    }
+    // returns the row of buttons
+    return (
+      <div className="flex flex-row justify-center mx-auto mt-auto mb-4">
+        {/* back button */}
+        <button
+          className="h-16 w-12 disabled:bg-gray-500 hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 border-l-2 border-t-2 border-b-2 border-black bg-white"
+          id="back-btn"
+          disabled={offset === 0}
+          onClick={handlePageClick}
+        >
+          {'<'}
+        </button>
+        {/* map buttons[] */}
+        {buttons.map(btn => btn)}
+        {/* next button */}
+        <button
+          className="h-16 w-12 disabled:bg-gray-500 hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300  border-r-2 border-t-2 border-b-2 border-black bg-white"
+          id="next-btn"
+          disabled={offset >= totalItems - itemsPerPage}
+          onClick={handlePageClick}
+        >
+          {'>'}
+        </button>
+      </div>
+    );
+  };
+  return (
+    <div className="max-w-screen-md">
+      <h3 className="mb-8 prose-sm">
+        Page {currentPageQuery}/{totalPages}
+      </h3>
+      <section>
+        {/* Component passed in that will render the data */}
+        {Component ? <Component currentItems={currentItems} /> : <RenderData />}
+      </section>
+      <div className="sticky lg:bottom-12 bottom-4">
+        <RenderButtons />
+      </div>
+    </div>
+  );
+};
+
+export default React.memo(Paginator);
```

Comment:
I'm not sure if it makes sense to memoize the whole component, but maybe pieces of the component like the buttons or the RenderedData itself

Reply:
Removed memoization of the whole component. I played around with profiler to test memoization on different pieces like the buttons and saw no performance increase so leaving memoization out of this component.